lib/Service/RetentionService.php

```diff
@@ -0,0 +1,358 @@
+		$reminderDayOptions = explode(',', $reminderDaysString);
+		foreach ($reminderDayOptions as $option) {
+			$option = (int) trim($option);
+			if ($option !== 0) {
+				$this->remindersPlain[] = $option;
+				$this->reminders[] = $now->sub(new \DateInterval('P' . $option . 'D'))->getTimestamp();
+			}
+		}
+
+		$this->keepUsersWithoutLogin = $this->config->getAppValue('user_retention', 'keep_users_without_login', 'yes') === 'yes';
+
+		try {
+			$excludedGroups = $this->config->getAppValue('user_retention', 'excluded_groups', '["admin"]');
+			$excludedGroups = json_decode($excludedGroups, true, 512, JSON_THROW_ON_ERROR);
+			$this->excludedGroups = \is_array($excludedGroups) ? $excludedGroups : [];
+		} catch (\JsonException $e) {
+		}
+
+		if ($this->keepUsersWithoutLogin) {
+			$this->userManager->callForSeenUsers(\Closure::fromCallable([$this, 'executeRetentionPolicy']));
+		} else {
+			$this->userManager->callForAllUsers(\Closure::fromCallable([$this, 'executeRetentionPolicy']));
+		}
+	}
+
+	public function executeRetentionPolicy(IUser $user): ?bool {
+		$this->logger->warning($user->getUID());
+		$skipIfNewerThan = $this->userMaxLastLogin;
+		$policyDays = $this->userDays;
+		if ($user->getBackend() instanceof UserBackend) {
+			$skipIfNewerThan = $this->guestMaxLastLogin;
+			$policyDays = $this->guestDays;
+		}
+
+		if (!$skipIfNewerThan) {
+			$this->logger->debug('Skipping retention because not defined for user backend: {user}', [
+				'user' => $user->getUID(),
+			]);
+			return true;
+		}
+
+		// Skip user completely when member of a protected group
+		try {
+			$this->skipUserBasedOnProtectedGroupMembership($user);
+		} catch (SkipUserException $e) {
+			$this->logger->debug($e->getMessage(), $e->getLogParameters());
+			return true;
+		}
+
+		// Check if we delete the user
+		try {
+			$this->shouldPerformActionOnUser($user, $skipIfNewerThan);
+
+			$this->logger->debug('Attempting to delete account: {user}', [
+				'user' => $user->getUID(),
+			]);
+			if($user->getBackendClassName() === 'LDAP' && !$this->prepareLDAPUser($user)) {
+				$this->logger->warning('Expired LDAP account ' . $user->getUID() . ' was not deleted');
+				return true;
+			}
+
+			if ($user->delete()) {
+				$this->logger->info('Account deleted: ' . $user->getUID());
+			} else {
+				$this->logger->warning('Expired account ' . $user->getUID() . ' was not deleted');
+			}
+			return true;
+		} catch (SkipUserException $e) {
+			// Not deleting yet, continue with checking reminders
+		}
+
+		// Check if we remind the user
+		foreach ($this->reminders as $key => $reminder) {
+			$reminderDays = $this->remindersPlain[$key] ?? 0;
+
+			$this->logger->debug('Checking reminder with {reminder} day: {user}', [
+				'reminder' => $reminderDays,
+				'user' => $user->getUID(),
+			]);
+
+			try {
+				$lastActivity = $this->shouldPerformActionOnUser($user, $reminder, $reminder - 86400);
+
+				// FIXME send notification
```

Comment:
this was done, to be removed

Reply:
Forgot to push?

Reply:
Nop, it's gone on the latest commit?

Reply:
Oh, I was looking for an actual notification to be sent, this was actually about sending the email itself?